dll/win32/shlwapi/propbag.cpp

```diff
@@ -1917,13 +1917,9 @@ SHGetViewStatePropertyBag(
         ::LeaveCriticalSection(&g_csBagCacheLock);
         return hr;
     }
-
-    g_pCachedBag.Attach(pBag);
-
-    hr = g_pCachedBag->QueryInterface(riid, ppv);
-
+    g_pCachedBag = pBag;
     ::LeaveCriticalSection(&g_csBagCacheLock);
-    return hr;
+    return pBag->QueryInterface(riid, ppv);
```

Comment:
why the change?

Reply:
Because we hold the lock for fewer instructions. Changing `Attach` to `=` fixed a crash and acting on the plain pBag interface outside the lock is safe.

Reply:
> and acting on the plain pBag interface outside the lock is safe.

But that pBag interface is now also what's now stored in the global `g_pCachedBag`, no?
So what would happen if in the meantime (just after the LeaveCriticalSection call and before the pBag->QueryInterface call), we get separate threads that call to _any_ of the other functions in dll/win32/shlwapi/propbag.cpp (those that acquire `g_csBagCacheLock` because they are going to act/read on the `g_pCachedBag`)

Reply:
There is no in-between LeaveCriticalSection and XYZ to worry about. The reference count of pBag is >= 1. CBasePropertyBag uses interlocked functions to handle the reference count...

Reply:
@katahiromz Any thoughts about this? (you wrote the code)